pkg/tests/ossm/prometheus_scoped_scraping_test.go

```diff
@@ -0,0 +1,275 @@
+package ossm
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/maistra/maistra-test-tool/pkg/app"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/common"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/require"
+	"github.com/maistra/maistra-test-tool/pkg/util/curl"
+	namespaces "github.com/maistra/maistra-test-tool/pkg/util/ns"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+)
+
+func TestOperatorCanUpdatePrometheusConfigMap(t *testing.T) {
+	test.NewTest(t).Groups(test.Full).Run(func(t test.TestHelper) {
+		t.Log("This test checks if the operator can update Prometheus ConfigMap when the SMMR is updated")
+
+		t.Cleanup(func() {
+			oc.ApplyString(t, meshNamespace, smmr)
+		})
+
+		prometheusPodSelector := pod.MatchingSelector("app=prometheus,maistra-control-plane=istio-system", meshNamespace)
+
+		checkPermissionErorr := func(t test.TestHelper) {
+			t.LogStep("Check the Prometheus log to see if there is any permission error")
+			oc.Logs(t,
+				prometheusPodSelector,
+				"prometheus",
+				assert.OutputDoesNotContain(
+					fmt.Sprintf("User \"system:serviceaccount:%s:prometheus\" cannot list resource", meshNamespace),
+					"Found no permission error",
+					"Expected to find no permission error, but got some error",
+				),
+			)
+		}
+
+		checkConfigurationReloadingTriggered := func(t test.TestHelper, start time.Time) {
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				oc.LogsSince(t,
+					start,
+					prometheusPodSelector, "config-reloader",
+					assert.OutputContains("Reload triggered",
+						"Triggered configuration reloading",
+						"Expected to trigger configuration reloading, but did not",
+					),
+				)
+			})
+		}
+
+		DeployControlPlane(t)
+
+		checkPermissionErorr(t)
+
+		getPrometheusConfigCmd := fmt.Sprintf("oc -n %s get configmap prometheus -o jsonpath='{.data.prometheus\\.yml}'", meshNamespace)
+
+		t.NewSubTest("when creating a SMMR").Run(func(t test.TestHelper) {
+			ns := generateNamespace()
+
+			t.Cleanup(func() {
+				oc.DeleteNamespace(t, ns)
+				restoreDefaultSMMR(t)
+			})
+
+			t.LogStepf("Create namespace %s and add it into SMMR", ns)
+			oc.CreateNamespace(t, ns)
+			updateDefaultSMMRWithNamespace(t, ns)
+
+			t.LogStepf("Look for %s in prometheus ConfigMap", ns)
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				shell.Execute(t, getPrometheusConfigCmd, checkForNamespace(ns))
+			})
```

Comment:
I am putting the `getPrometheusConfigCmd` into [`testPrometheusConfigWithAsserts`](https://github.com/maistra/maistra-test-tool/pull/440/commits/59d4812f6e5ea5bad5c99f79987bb18bad2f88b1#diff-bfccbb25857d3619f1b88999f773a7b30f5afe3aa577912be7c79c04882b7b19R58). I did not use assert in the function name, because the function takes a `TestHelper` and I want to differentiate it from the `CheckFunc`.